### ISSUE TITLE
CT should not be called if disposed

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/expression-exchange#147]
```
expression-exchange : fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/expression-exchange/expression-exchange_en.html?continuousTest=%7B%22test%22%3A%5B%22expression-exchange%22%2C%22fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1604715218212%22%2C%22timestamp%22%3A1604846732443%7D&brand=phet&ea&fuzz&memoryLimit=1000
Query: brand=phet&ea&fuzz&memoryLimit=1000
Uncaught Error: Assertion failed: should not be called if disposed
Error: Assertion failed: should not be called if disposed
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/assert/js/assert.js:22:13)
at TinyForwardingProperty.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyEmitter.js:53:15)
at TinyForwardingProperty.notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyForwardingProperty.js:178:10)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyProperty.js:70:12)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyForwardingProperty.js:162:13)
at VariableCoinTermNode.setPickable (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/scenery/js/nodes/Node.js:3464:28)
at VariableCoinTermNode.set pickable [as pickable] (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/scenery/js/nodes/Node.js:3468:32)
at updatePickability (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/expression-exchange/js/common/view/AbstractCoinTermNode.js:113:19)
at TinyEmitter.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyEmitter.js:71:18)
at Property._notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/Property.js:280:25)
id: Bayes Chrome
Snapshot from 11/6/2020, 7:13:38 PM

----------------------------------

expression-exchange : fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/expression-exchange/expression-exchange_en.html?continuousTest=%7B%22test%22%3A%5B%22expression-exchange%22%2C%22fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1604715218212%22%2C%22timestamp%22%3A1604908767126%7D&brand=phet&ea&fuzz&memoryLimit=1000
Query: brand=phet&ea&fuzz&memoryLimit=1000
Uncaught Error: Assertion failed: should not be called if disposed
Error: Assertion failed: should not be called if disposed
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/assert/js/assert.js:22:13)
at TinyForwardingProperty.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyEmitter.js:53:15)
at TinyForwardingProperty.notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyForwardingProperty.js:178:10)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyProperty.js:70:12)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyForwardingProperty.js:162:13)
at VariableCoinTermNode.setPickable (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/scenery/js/nodes/Node.js:3464:28)
at VariableCoinTermNode.set pickable [as pickable] (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/scenery/js/nodes/Node.js:3468:32)
at updatePickability (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/expression-exchange/js/common/view/AbstractCoinTermNode.js:113:19)
at TinyEmitter.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/TinyEmitter.js:71:18)
at Property._notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1604715218212/axon/js/Property.js:280:25)
id: Bayes Chrome
Snapshot from 11/6/2020, 7:13:38 PM
```

**Comment by jbphet** [comment]
This isn't currently showing up on CT (see screenshot below).  My guess is that this was a temporary problem related to the work being done for the general support of enabled properties.  I'm going to close, and if it comes back we can re-open or create a new one.

![image](https://user-images.githubusercontent.com/3082831/98607909-54a64e00-22a7-11eb-8436-c31e5d565a38.png)

**Comment by KatieWoe** [comment]
Showed up again, reopening:
```
expression-exchange : fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/expression-exchange/expression-exchange_en.html?continuousTest=%7B%22test%22%3A%5B%22expression-exchange%22%2C%22fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1611503573321%22%2C%22timestamp%22%3A1611514505954%7D&brand=phet&ea&fuzz&memoryLimit=1000
Query: brand=phet&ea&fuzz&memoryLimit=1000
Uncaught Error: Assertion failed: should not be called if disposed
Error: Assertion failed: should not be called if disposed
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/assert/js/assert.js:25:13)
at TinyForwardingProperty.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/axon/js/TinyEmitter.js:53:15)
at TinyForwardingProperty.notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/axon/js/TinyForwardingProperty.js:178:10)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/axon/js/TinyProperty.js:70:12)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/axon/js/TinyForwardingProperty.js:162:13)
at VariableCoinTermNode.setPickable (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/scenery/js/nodes/Node.js:3453:28)
at VariableCoinTermNode.set pickable [as pickable] (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/scenery/js/nodes/Node.js:3457:32)
at updatePickability (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/expression-exchange/js/common/view/AbstractCoinTermNode.js:112:21)
at TinyEmitter.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/axon/js/TinyEmitter.js:71:18)
at Property._notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1611503573321/axon/js/Property.js:280:25)
id: Bayes Chrome
Snapshot from 1/24/2021, 8:52:53 AM
```

**Comment by jbphet** [comment]
I've updated the drag listeners and the drag forwarding to the latest versions, and CT has been clear for over 24 hours, so I'm going to call this good and close.

**Comment by jbphet** [comment]
Dang.  This issue appears to have occurred again over the weekend.  Here is the stack trace:

```
expression-exchange : fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/expression-exchange/expression-exchange_en.html?continuousTest=%7B%22test%22%3A%5B%22expression-exchange%22%2C%22fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1615064572807%22%2C%22timestamp%22%3A1615092779553%7D&brand=phet&ea&fuzz&memoryLimit=1000
Query: brand=phet&ea&fuzz&memoryLimit=1000
Uncaught Error: Assertion failed: should not be called if disposed
Error: Assertion failed: should not be called if disposed
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/assert/js/assert.js:25:13)
at TinyForwardingProperty.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/axon/js/TinyEmitter.js:65:15)
at TinyForwardingProperty.notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/axon/js/TinyForwardingProperty.js:179:10)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/axon/js/TinyProperty.js:71:12)
at TinyForwardingProperty.set (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/axon/js/TinyForwardingProperty.js:163:13)
at VariableCoinTermNode.setPickable (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/scenery/js/nodes/Node.js:4128:28)
at VariableCoinTermNode.set pickable [as pickable] (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/scenery/js/nodes/Node.js:4140:10)
at updatePickability (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/expression-exchange/js/common/view/AbstractCoinTermNode.js:111:21)
at TinyEmitter.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/axon/js/TinyEmitter.js:86:18)
at Property._notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1615064572807/axon/js/Property.js:294:25)
id: Bayes Chrome
Snapshot from 3/6/2021, 2:02:52 PM
```

**Comment by jbphet** [comment]
I did a lot of local fuzz testing after implementing a fix for #151, and never saw this error.  I'm not sure whether this means the problem is exceedingly rare, or that the fixes for that issue also addressed this one.  I'm going to close, but will re-open if it crops up again.  If so, this may be the record for the issue with the most closes and re-opens. :-)

**Comment by KatieWoe** [comment]
Sorry @jbphet 
```
expression-exchange : fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1654292254278/expression-exchange/expression-exchange_en.html?continuousTest=%7B%22test%22%3A%5B%22expression-exchange%22%2C%22fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1654292254278%22%2C%22timestamp%22%3A1654322249519%7D&brand=phet&ea&fuzz&memoryLimit=1000
Query: brand=phet&ea&fuzz&memoryLimit=1000
Uncaught Error: Assertion failed: should not be called if disposed
Error: Assertion failed: should not be called if disposed
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1654292254278/assert/js/assert.js:28:13)
at assert (TinyEmitter.ts:73:14)
at emit (TinyProperty.ts:130:9)
at notifyListeners (TinyProperty.ts:78:11)
at set (TinyForwardingProperty.ts:138:12)
at set (Node.ts:4125:27)
at setPickable (Node.ts:4134:9)
at (AbstractCoinTermNode.js:112:6)
at listener (AbstractProperty.ts:361:4)
at link (AbstractCoinTermNode.js:126:38)
id: Bayes Puppeteer
Snapshot from 6/3/2022, 3:37:34 PM
```

**Comment by jbphet** [comment]
I'd really love to understand how and why this happens, but I tried fuzzing for about 10 minutes (again) and never hit it.  It is apparently just a very rare condition.  It doesn't seem like it's worth much more time, so I just added a guard to the code to prevent it from trying to change the pickability if the coin term node has been disposed.  I think that should prevent it from ever happening again.  I'll close this one, but will keep an eye on CT and re-open if I see anything related.
